### PR TITLE
Add image and video direct link parsing and download size limit

### PR DESCRIPTION
**Pull request by novill**

### CODE REVIEW

File: lib/link_thumbnailer/processor.rb [hunk 1]
```diff
@@ -103,19 +116,26 @@ def too_many_redirections?
...
-      false
+      if response['Content-Type'] =~ (/text\/html|application\/html|application\/xhtml\+xml|application\/xml|text\/xml|text\/plain/)
+        @content_type=:application
+        return true
+      elsif response['Content-Type'] =~ /image\//
+        @content_type=:image
+        return true
+      elsif response['Content-Type'] =~ /video\//
+        @content_type=:video
+        return tru
```
Comment (gottfrois) [inline lib/link_thumbnailer/processor.rb]: `true`

**Comment by maia** [comment]
@gottfrois Would you consider integrating the changes regarding limiting the download file size? It seems all it needs is a change in `Processor#perform_request` reading the body in chunks (description of Net::HTTPResponse reads: "If a block is given, the body is passed to the block, and the body is provided in fragments, as it is read in from the socket."), plus a configuration option and default value. Or do you see any additional necessity to solve the issue in typhoeus, as you mentioned in #80 ? Thanks!

**Comment by gottfrois** [comment]
@maia you mean switch from typhoeus to net http?

**Comment by maia** [comment]
@gottfrois I actually don't really know how and where typhoeus is used, as I couldn't find any references to `Typhoeus::Request` in the code. Without being very familiar with the code, `Processor#perform_request` uses the instance variable `@http` which is of the class `Net::HTTP::Persistent`. And this probably wraps net http, so the currently used `response = http.request(url)` uses net http as the suggested change to `response = http.request(url) {|resp| resp.read_body {|chunk|} … }` would also do – meaning, I see nothing in this part of the code related to typhoeus. Even if you're using typhoeus in other places of the code, wouldn't this change allow to limit the downloaded file size? Or what am I missing? 

I'd happily assist in solving the issue, but currently I do not understand why this seemingly easy change does not seem to be the solution or is not the direction you'd like to take the gem.

The reason I'm asking if this could be a solution is that I run an app that parses urls in a background job with a bunch of parallel workers, and sometimes I'm seeing terrible memory increases most likely caused by servers sending a wrong `content-length` header, and the job then loads a file much larger than acceptable for the environment. Currently I limit the workers to leave 100MB headroom on the 512MB dyno, but I'd prefer to run more workers and prevent sudden spikes in used memory.

**Comment by gottfrois** [comment]
Ok I recall now. Typhoeus is used on the [image_info](https://github.com/gottfrois/image_info) gem which is responsible to DL images.
I'll be happy to merge the proposed change here however in its current state, this PR is conflicting with latest master and failing specs.

@maia I'll be glad to merge a workable PR from you if you feel like doing it.

**Comment by maia** [comment]
@gottfrois sure, I'll provide a PR this week.

**Pull request closed without merging**